src/drawable.typ

```diff
@@ -3,6 +3,19 @@
 #import "util.typ"
 #import "path-util.typ"
 
+#let apply-transform-segments(transform, segments) = {
+  if transform == none {
+    return segments
+  }
+
+  assert(type(segments) == array,
+    message: "Expected segment array, got " + repr(segments))
+  return segments.map(segment => {
+    let (kind, ..pts) = segment
+    return (kind,) + util.apply-transform(transform, ..pts)
+  })
+}
```

Comment:
Would this be better in path-utils?

Because this function is used in a loop in `apply-transform`, I would move lines 7 to 12 to another function which then calls this one. So the transform if statement isn't done and segments assert is not done every loop.
```
let apply-transform-segments-unchecked(transform, segments) = {...}
let apply-transform-segments-checked(transform, segments) = {
  if transform ...
  assert(...)
  apply-transform-unchecked(...)
}

let apply-transform(...) = {
  ...
  if transform ...
  for drawable in drawables {
    ...
    apply-transform-segments-unchecked(transform, drawables.segments)
  }
```
Although at this point it starts to become a mess...

Reply:
Removed.